optuna/samplers/_cmaes.py

```diff
@@ -242,30 +283,46 @@ def sample_relative(
         study._storage.set_trial_system_attr(
             trial._trial_id, "cma:generation", optimizer.generation
         )
+        study._storage.set_trial_system_attr(trial._trial_id, "cma:n_restarts", n_restarts)
         external_values = {
             k: _to_optuna_param(search_space[k], p) for k, p in zip(ordered_keys, params)
         }
         return external_values
 
-    def _restore_or_init_optimizer(
-        self,
-        completed_trials: "List[optuna.trial.FrozenTrial]",
-        search_space: Dict[str, BaseDistribution],
-        ordered_keys: List[str],
-    ) -> CMA:
-
+    def _restore_optimizer(
+        self, completed_trials: "List[optuna.trial.FrozenTrial]",
+    ) -> Tuple[Optional[CMA], int]:
         # Restore a previous CMA object.
         for trial in reversed(completed_trials):
             serialized_optimizer = trial.system_attrs.get(
                 "cma:optimizer", None
             )  # type: Optional[str]
             if serialized_optimizer is None:
                 continue
-            return pickle.loads(bytes.fromhex(serialized_optimizer))
+            n_restarts = trial.system_attrs.get("cma:n_restarts", 0)  # type: int
+            return pickle.loads(bytes.fromhex(serialized_optimizer)), n_restarts
+        return None, 0
 
-        # Init a CMA object.
-        if self._x0 is None:
-            self._x0 = _initialize_x0(search_space)
+    def _init_optimizer(
+        self,
+        search_space: Dict[str, BaseDistribution],
+        ordered_keys: List[str],
+        population_size: Optional[int] = None,
+        n_restarts: int = 0,
+    ) -> CMA:
+        if n_restarts > 0:
+            # `_initialize_x0_uniformly` returns internal representations.
+            x0 = _initialize_x0_uniformly(self._cma_rng, search_space)
+            mean = np.array([x0[k] for k in ordered_keys], dtype=float)
+        elif self._x0 is None:
+            # `_initialize_x0` returns internal representations.
+            x0 = _initialize_x0(search_space)
+            mean = np.array([x0[k] for k in ordered_keys], dtype=float)
+        else:
+            # `self._x0` is external representations.
+            mean = np.array(
+                [_to_cma_param(search_space[k], self._x0[k]) for k in ordered_keys], dtype=float
+            )
```

Comment:
The users set external representations at `x0` argument of the constructor. So we need to convert it to cma_param. So previous logic seems to be broken...

`_initialize_x0` and `_initialize_x0_uniformly` returns its internal representations as I commented https://github.com/optuna/optuna/pull/1548/files#r463950014.